File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;
import java.io.*;

public class Solution {
    public int[] solution(int []arr) {
        Deque<Integer> deque = new LinkedList<>();
        
        for(int i = 0; i < arr.length; i++){
            if(i > 0 && arr[i] == arr[i-1]){
                continue;
            }
            deque.addFirst(arr[i]);  
        }
        
        int size = deque.size();
        int[] answer = new int[deque.size()];
        for(int i = 0; i < size; i++){
            answer[i] = deque.removeLast();
        }

        return answer;
    }
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.js
function solution(arr)
{
    let stack = [];
    for(let n of arr){
        if(stack.length === 0){
            stack.push(n);
            continue;
        }
        
        let pop = stack.pop();
        if (n !== pop){
            stack.push(pop);
            stack.push(n);
        } else {
            stack.push(pop);
        }
    }
    
    return stack;
}